exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetScanBatchCreator.java

```diff
@@ -156,18 +160,39 @@ public ScanBatch getBatch(FragmentContext context, ParquetRowGroupScan rowGroupS
     return new ScanBatch(rowGroupScan, context, oContext, readers, implicitColumns);
   }
 
-  private static boolean isComplex(ParquetMetadata footer) {
-    MessageType schema = footer.getFileMetaData().getSchema();
+  private static boolean isComplex(ParquetMetadata footer, List<SchemaPath> columns) {
+    if (Utilities.isStarQuery(columns)) {
```

Comment:
Perhaps a comment with some explanation? If wildcard query, we query all columns, so check if any of them are complex. If project list, then check only the projected columns.

Reply:
Added in a new commit.